shared/domains/chelonia/chelonia.js

```diff
@@ -48,13 +55,43 @@ export const ACTION_REGEX: RegExp = /^((([\w.]+)\/([^/]+))(?:\/(?:([^/]+)\/)?)?)
 // 4 => 'group'
 // 5 => 'payment'
 
+const signatureFnBuilder = (key) => {
+  return (data) => {
+    return {
+      type: key.type,
+      keyId: keyId(key),
+      data: sign(key, data)
+    }
+  }
+}
+
 sbp('sbp/selectors/register', {
   // https://www.wordnik.com/words/chelonia
   // https://gitlab.okturtles.org/okturtles/group-income/-/wikis/E2E-Protocol/Framework.md#alt-names
   'chelonia/_init': function () {
     this.config = {
-      decryptFn: JSON.parse, // override!
-      encryptFn: JSON.stringify, // override!
+      decryptFn: function (message: Object, state: ?Object) {
+        if (Object(message) instanceof String) {
+          return JSON.parse(message)
+        }
+
+        const keyId = message.keyId
+        const key = this.env.additionalKeys?.[keyId] || state?._volatile?.keys?.[keyId]
+
+        return JSON.parse(decrypt(key, message.content))
+      },
+      encryptFn: function (message: Object, eKeyId: string, state: ?Object) {
+        const key = this.env.additionalKeys?.[eKeyId] || state?._volatile?.keys?.[eKeyId]
+
+        if (!key) {
+          return JSON.stringify(message)
+        }
+
+        return {
+          keyId: keyId(key),
+          content: encrypt(key, JSON.stringify(message))
+        }
+      },
```

Comment:
There's a certain aesthetic we're aiming for, so whenever gnarliness starts showing up we will tend to create new functions to keep things easy to read. Could you create two functions down below next to `outEncryptedOrUnencryptedAction` and reference that here to keep `_init` easy to read?